### PR TITLE
Pretty print using prettier 

### LINKED ISSUE #18: Pretty output: run the program through prettier
TypeScript printer is not printing the most beautiful code. We can run the code through prettier as second phase of processing to get nicer output.

### PR DESCRIPTION
**Pull request by mohsen1**
Fixes #18 
# TODO:

- [x] Update tests with pretty printed output
- [x] Detect single quote vs double quote from source file and use it for prettier 
- [x] Detect indentation from source file and use it for prettier 
- [x] Detect code width from source file and use it for prettier 
- [x] Detect semi from source file and use to for prettier option (not sure if possible)
- [x] **CLI** accept all prettier options and proxy it to prettier

### CODE REVIEW

File: test/collapse-intersection-interfaces-transform/simple/output.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
 type Foo = {
-    foo: string;
-    bar: number;
+foo: string,
```
Comment (mohsen1) [inline test/collapse-intersection-interfaces-transform/simple/output.tsx]: Why no indent?!

File: test/end-to-end/non-react/output.tsx [hunk 2]
```diff
@@ -1,9 +1,8 @@
 class Foo {
     render() {
-        return '100';
+        return '100'
```
Comment (mohsen1) [inline test/end-to-end/non-react/output.tsx]: semi should be on by default. this is a bug

File: test/react-stateless-function-make-props-transform/stateless-function/output.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
 type HelloProps = {
-    message?: string;
+	message?: string
```
Comment (mohsen1) [inline test/react-stateless-function-make-props-transform/stateless-function/output.tsx]: tabs?!

**Comment by vincentbel** [comment]
@mohsen1 I think we can resolve prettier config by default by using [`prettier.resolveConfig`](https://prettier.io/docs/en/api.html#prettierresolveconfigfilepath-options). And also support setting prettier config in CLI.

**Comment by mohsen1** [comment]
Beautiful! I didn't notice that api

**Pull request merged**